tree/tree/src/TBasket.cxx

```diff
@@ -617,19 +639,33 @@ Int_t TBasket::ReadBasketBuffers(Long64_t pos, Int_t len, TFile *file)
    fBranch->GetTree()->IncrementTotalBuffers(fBufferSize);
 
    // Read offsets table if needed.
-   if (!fBranch->GetEntryOffsetLen()) {
+   // If there's no EntryOffsetLen in the branch -- or the fEntryOffset is marked to be calculated-on-demand --
+   // then we skip reading out.
+   if (!fBranch->GetEntryOffsetLen() || (fEntryOffset == reinterpret_cast<Int_t *>(1))) {
       return 0;
    }
-   delete [] fEntryOffset;
-   fEntryOffset = 0;
+   // At this point, we're required to read out an offset array.
+   ResetEntryOffset(); // TODO: every basket, we reset the offset array.  Is this necessary?
+                       // Could we instead switch to std::vector?
    fBufferRef->SetBufferOffset(fLast);
    fBufferRef->ReadArray(fEntryOffset);
-   if (!fEntryOffset) {
+   if (R__unlikely(!fEntryOffset)) {
       fEntryOffset = new Int_t[fNevBuf+1];
       fEntryOffset[0] = fKeylen;
       Warning("ReadBasketBuffers","basket:%s has fNevBuf=%d but fEntryOffset=0, pos=%lld, len=%d, fNbytes=%d, fObjlen=%d, trying to repair",GetName(),fNevBuf,pos,len,fNbytes,fObjlen);
       return 0;
    }
+   if (fIOBits & static_cast<UChar_t>(TBasket::EIOBits::kGenerateOffsetMap)) {
+      // In this case, we cannot regenerate the offset array at runtime -- but we wrote out an array of
+      // sizes instead of offsets (as sizes compress much better).
+      fEntryOffset[0] = fKeylen;
+      for (Int_t idx = 1; idx < fNevBuf + 1; idx++) {
+         printf("Reading offset for idx %d; old value %d", idx, fEntryOffset[idx]);
```

Comment:
of course for adding to the master, those would need to be at least commented out :)

Reply:
Shoot.  No matter how many times you check, one always slips through... will fix.